### PR TITLE
FQCN roles

### PR DESCRIPTION
**Pull request by thekad**
Roles referencing other collection roles should also FQCN their include/import

Test-Hints: assisted

### CODE REVIEW

File: roles/boot_disk/tasks/main.yml
Comment (tonyskapunk) [inline roles/boot_disk/tasks/main.yml:11]: I'm not sure this is correct, I think it should use FQCN:

```suggestion
    name: "redhatci.ocp.vendors.{{ hostvars[item]['vendor'] | lower }}"
```

File: roles/hco_setup/README.md
Comment (tonyskapunk) [inline roles/hco_setup/README.md:40]: Let's use `ansible.builtin.include_role` as in the other READMEs

File: roles/install_cluster/tasks/main.yml
Comment (tonyskapunk) [inline roles/install_cluster/tasks/main.yml:25]: Should we  use `ansible.builtin.include_role` here?

File: roles/merge_registry_creds/README.md
Comment (tonyskapunk) [inline roles/merge_registry_creds/README.md:21]: `ansible.builtin.include_role` ?

**Review by tonyskapunk — CHANGES_REQUESTED** [review]

File: roles/boot_disk/tasks/main.yml
Comment (thekad) [reply roles/boot_disk/tasks/main.yml:11]: I am not sure that's correct either, but your suggestion seems better

**Review by thekad — COMMENTED** [review]
Most of the errors came from a stashed chunk I applied before my other PR got merged, so I had a regression on some of my FQCN includes before. I have updated the READMEs with your suggestions as well

File: roles/validate_inventory/tasks/vendor.yml
Comment (tonyskapunk) [inline roles/validate_inventory/tasks/vendor.yml:12]: Could you also use FQCN in the include_role? also the following files don't have FQCN in the include_role let's change them too

roles/generate_discovery_iso/tasks/static.yml
roles/add_day2_node/tasks/main.yml

Thanks

Comment (thekad) [reply roles/validate_inventory/tasks/vendor.yml]: all roles coming from crucible need to be audited for FQCN, but I'm not doing that just now. I have changed the files you mentioned, but let's hold off on the rest for another PR.

Comment (tonyskapunk) [reply roles/validate_inventory/tasks/vendor.yml]: Thanks! I was referring only to the include_role part :smile: but I see you have changed them all, sorry if I was not clear.

**Review by tonyskapunk — APPROVED** [review]
LGTM

I see now the sub-roles are listed with FQCN https://www.distributed-ci.io/jobs/3bc93b41-0acf-4223-ab03-2f42990c48b3/jobStates?sort=date&task=19132701-0fbb-4f7a-a102-0982bf14e741

File: roles/boot_iso/tasks/main.yml
Comment (nocturnalastro) [inline roles/boot_iso/tasks/main.yml:12]: I will have to check this still works.

File: roles/install_cluster/tasks/main.yml
Comment (nocturnalastro) [inline roles/install_cluster/tasks/main.yml:213]: Note sure if there should be triple dots here. Is this a yaml thing?

Comment (thekad) [reply roles/install_cluster/tasks/main.yml:213]: yeah this is just yaml syntax for "end of document"

File: roles/boot_iso/tasks/main.yml
Comment (nocturnalastro) [reply roles/boot_iso/tasks/main.yml:12]: Worked fine for me using a vendor kvm.

**Review by nocturnalastro — APPROVED** [review]

**Review by nocturnalastro — APPROVED** [review]

**Review by nocturnalastro — APPROVED** [review]

**Pull request merged**
